config/environments/test.rb

```diff
@@ -44,6 +44,10 @@
   # Store test uploads on the disk in the temp folder
   config.active_storage.service = :test
 
+  config.action_mailer.delivery_method = :test
+  config.action_mailer.perform_deliveries = false
+  config.action_mailer.default_url_options = {host: "your-host-name"}
+
```

Comment:
This is added because my test is creating user, and creating user is sending email which is causing error

Reply:
We shouldn't need to create users under test just yet. You should be able to test your code with `users(:with_available_profile)` vs. `users(:without_profile)`.